packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo } from 'react';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
+
+import { mergeClasses } from '../../classify';
+import CategorySortItem from './categorySortItem';
+import defaultClasses from './categorySort.css';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+    const { elementRef, expanded, setExpanded } = useDropdown();
+
+    const sortElements = useMemo(() => {
+        const onMenuItemClick = sortAttribute => {
+            setSort({
+                sortAttribute: sortAttribute.attribute,
+                sortDirection: sortAttribute.sortDirection
+            });
+            setExpanded(false);
+        };
+
+        // should be not render item in collapsed mode.
+        if (expanded === false) {
+            return;
+        }
+
+        const sortElements = [];
+        for (const element of availableSortMethods) {
+            const attribute = element.attribute;
+            const sortDirection = element.sortDirection;
+            const isActive =
+                currentSort.sortAttribute === attribute &&
+                currentSort.sortDirection === sortDirection;
+
+            const key = `${attribute}--${sortDirection}`;
+            sortElements.push(
+                <li key={key} className={classes.menuItem}>
+                    <CategorySortItem
+                        sortItem={element}
+                        active={isActive}
+                        onClick={function(element) {
+                            onMenuItemClick(element);
+                        }}
```

Comment:
When you pass an inline function to `onClick`, it will be recreated on every render. That's especially not optimal since it has to be rebound in the DOM.

The general rule here is to never create a "version" of the handler in the parent and pass it down to the child. Instead, pass the necessary data (and any static handlers) to the child and let it compose its own handler.

```jsx
// parent
const List = props => {
  // memoize to avoid recreating this function on render
  // make sure its deps aren't going to change
  const handleItemClick = useCallback(() => {
    setExpanded(false)
  }, [setExpanded])

  // iterate over data to create elements
  // note that we don't define any handlers in the loop
  const itemElements = Array.from(data, item => {
    return (
      <Item item={item} onClick={handleItemClick} />
    )
  })
}

// child
const Item = props => {
  const { item } = props
  const { foo } = item

  // define a handler at the item level, not the list level
  const handleClick = useCallback(() => {
    // do something with the data we received as a prop
    console.log(foo)

    // if we accept a callback, call it here
    // note that received callbacks are `onEvent`
    // while defined callbacks are `handleEvent`
    onClick()
  }, [foo, onClick])

  return <button onClick={handleClick} />
}
```